libs/execution_context/ebpf_maps.c

```diff
@@ -264,6 +265,26 @@ _associate_program_with_prog_array_map(_In_ ebpf_core_map_t* map, _In_ const ebp
     return result;
 }
 
+static bool // Returns true if ok, false if not.
+_check_value_type(_In_ const ebpf_core_map_t* outer_map, _In_ const ebpf_object_t* value_object)
```

Comment:
Maybe add an explanation that this function releases a reference on inner_map_template?

Reply:
It shouldn't, I see a bug in the caller that will leak a reference for PROG_ARRAY, let me clean this up.

Reply:
Actually it shouldn't, there was a bug in my code which I have now fixed.